timemachine/cpp/src/fixed_point.hpp

```diff
@@ -2,6 +2,16 @@
 
 #define FIXED_EXPONENT 0x1000000000
 
+// we need to use a different level of precision for parameter derivatives
+#define FIXED_EXPONENT_DU_DCHARGE 0x1000000000
+#define FIXED_EXPONENT_DU_DSIG 0x2000000000
+#define FIXED_EXPONENT_DU_DEPS 0x4000000000 // this is just getting silly
```

Comment:
Naive question: Are there conditions when this list of exponents might need to change in the future?

(E.g. would we expand the list for other parameter types, or would the numeric values need to be adjusted if some assumption changed? Not sure if these currently assume we're going to call `du_dparams(x, params)` only when `u(x, params)` is in some well-behaved range -- perhaps the exponents would need to be different if we start calling `du_dparams` on highly "off-equilibrium" `(x, params)`?)

Reply:
Hmm, it seems likely that we might want to change these at some point (maybe @proteneer will have a better idea). To be clear, this PR only pushes the existing conversions to the outer layer (the wrapper in `wrap_kernels.cpp`), so I think it should be mostly orthogonal to that issue. Might be worth opening another issue for discussion?

Reply:
Let's not change those values, but document the dynamic range of the supported precision.

Reply:
Should we delegate this to #481?

Reply:
Moved!